### DISCUSSION TITLE
Translation for account number bringing back English mix

### DISCUSSION THREAD

**Discussion by tsmith-ng** [openai/whisper#2226] [Q&A]
I have audio of users giving me thier account number which is a mix of letters and numbers. For the most part, audio.translation does fine when they are spoken in English. But when I try audio in French, it is all over the place.  Funny that translation API doesn't take language! 

Example: 12BB99 (une deux "bae" "bae" neuf neuf)
Returned: 12baby99

What am I doing wrong? Thanks!

**Comment by glangford** [comment]
whisper will try to identify the language based on the first 30s of audio (if you don't specify the language). But it isn't always correct.

If you know the language of the audio is French, try specifying `language='fr'`. If you don't know the language, the answer here might be helpful:

* https://github.com/openai/whisper/discussions/1982

You could also try using `initial_prompt` and/or a larger model.
https://cookbook.openai.com/examples/whisper_prompting_guide

**Reply by tsmith-ng** [reply]
Thanks! For the translation API there isn't a language property. You can only set that on the transcribe method. I will take a look at the initial_prompt though.

**Reply by tsmith-ng** [reply]
What is also strange is that it properly translates the 1 2 and then messes up the letters and then gets the 9 9 right.

**Reply by tsmith-ng** [reply]
It could be hearing the BB as "bébé" which is French for "baby"
